onnx/defs/schema.cc

```diff
@@ -262,7 +262,7 @@ void OpSchema::Verify(const NodeProto& node) const {
 
     // Type would be UNDEFINED if not set
     if (attr_proto.type() != expected_type) {
-      fail_check("Mismatched attribute type in '", node.name() + " : " + name, "'");
+      fail_check("Mismatched attribute type in '", node.name() + " : " + name, "'. Expected: '", expected_type, "', actual: '", attr_proto.type(), "'");
```

Comment:
Is there a to string method for the attribute types?

Reply:
Protobuf generates a [function](https://protobuf.dev/reference/cpp/cpp-generated/#enum). So, it should be something like `AttributeType_Name(...)`. You may have to find the right qualified name to use for this nested type in proto.

Reply:
Ok, here is a usage: https://github.com/onnx/onnx/blob/b60f69412abb5393ab819b936b473f83867f6c87/onnx/defs/parser.cc#L559

Reply:
Done